api/hypershift/v1beta1/nodepool_types.go

```diff
@@ -751,6 +754,20 @@ type KubevirtNetwork struct {
 	Name string `json:"name"`
 }
 
+type KubevirtGPU struct {
+	// DeviceName is the name of the GPU device (either passthrough or vGPU) that is desired to be
+	// utilized in the HostedCluster's NodePool
+	DeviceName string `json:"deviceName"`
+
+	// Count is the number of instances the specified GPU device will be attached to each of the
+	// NodePool's nodes. Default is 1.
+	//
+	// +optional
+	// +kubebuilder:default=1
+	// +kubebuilder:validation:Minimum=1
+	Count int `json:"count,omitempty"`
+}
+
```

Comment:
I don't think this will conflict with future usage of "Dynamic Resource Allocation". It's possible that this struct will get expanded in the future to allow us to express resourceClaims here as well.

The only thing i'm slightly questioning is whether we should call this struct `KubevirtHostDevices` or `KubevirtGPUs`. I'm starting a slack conversation in `#hypershift-kubevirt-platform` to talk through that possiblity

Reply:
We talked about this out of band a bit already, but i wanted to leave the note on the PR so there's some history of this discussion.

I think we call this struct `KubevirtHostDevices` instead of `KubevirtGPU`. This gives us the ability to support more devices on the ndoepool outside of just GPUs in the future.